From: Andreas Hasenack <andreas.hasenack@canonical.com>
Date: Fri, 3 Apr 2026 15:15:13 +0800
Subject: drop deprecation warning in pam_env

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1018106
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/2059859
Forwarded: no
Last-Update: 2024-04-10

In version 1.5.0[1], PAM upstream added a deprecation warning[2] keyed on the
user_readenv option, signaling that the feature of reading user environment
files will be removed in the future. This was brought to Ubuntu's attention
via bug LP: #2059859[3], but since we were inside Beta Freeze for Ubuntu Noble
24.04 already, removing the user_readenv=1 option we add to sshd's PAM config
would introduce the kind of change in behavior that is not allowed during a
Beta Freeze.

Since such a change will also never be allowed in a Stable Release Update
(SRU), we decided to remove the deprecation warning for Ubuntu Noble 24.04 to
avoid the unecessary warning to our users.

This reverts commit[2], bar the NEWS and manpage changes.

After the 24.04 release, i.e., for 24.10 and later, this patch should be
removed and the openssh PAM configuration changed instead.

1. https://github.com/linux-pam/linux-pam/releases/tag/v1.5.0
2. https://github.com/linux-pam/linux-pam/commit/ecd526743a27157c5210b0ce9867c43a2fa27784
3. https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/2059859
---
 modules/pam_env/pam_env.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/modules/pam_env/pam_env.c b/modules/pam_env/pam_env.c
index 5947b86..44b29fa 100644
--- a/modules/pam_env/pam_env.c
+++ b/modules/pam_env/pam_env.c
@@ -141,9 +141,6 @@ _pam_parse (const pam_handle_t *pamh, int argc, const char **argv,
 	  pam_syslog(pamh, LOG_ERR, "unknown option: %s", *argv);
     }
 
-    if (*user_readenv)
-	pam_syslog(pamh, LOG_DEBUG, "deprecated reading of user environment enabled");
-
     return ctrl;
 }
 
